include/llbuild/BuildSystem/BuildSystemFrontend.h

```diff
@@ -160,6 +160,16 @@ class BuildSystemFrontendDelegate : public BuildSystemDelegate {
   /// corresponding \see commandFinished() call.
   virtual void commandPreparing(Command*) override;
 
+  /// Called by the build system to allow the delegate to skip a command.
+  ///
+  /// This method is called before the command starts, when the system has
+  /// identified that it will eventually need to run (after all of its inputs
+  /// have been satisfied).
+  ///
+  /// The system guarantees that all such calls will be paired with a
+  /// corresponding \see commandFinished() call.
```

Comment:
I think we should put a honkin' big warning in here about how this will break the consistency of the build and you should only use it if you know what you are doing. :)